pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test: TestLogic_refcursor failed

pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test.TestLogic_refcursor [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12497573?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12497573?buildTab=artifacts#/) on master @ [0496181c886c305f261598f90f0b226f9940fdac](https://github.com/cockroachdb/cockroach/commits/0496181c886c305f261598f90f0b226f9940fdac):


```
pkg/testutils/physicalplanutils/fake_resolver.go:28: (FakeResolverForTestCluster)
	NOTICE: .AdvRPCAddr() called via implicit interface ApplicationLayerInterface;
HINT: consider using .ApplicationLayer().AdvRPCAddr() instead.
TIP: consider replacing the test server initialization from:
    ts, ... := serverutils.StartServer(t, ...)
    defer ts.Stopper().Stop(...)
to:
    srv, ... := serverutils.StartServer(t, ...)
    defer srv.Stopper().Stop(...)
    ts := srv.ApplicationLayer()

See also: https://go.crdb.dev/p/testserver-and-cluster-virtualization

pkg/testutils/physicalplanutils/fake_resolver.go:27: (FakeResolverForTestCluster)
	NOTICE: .NodeID() called via implicit interface StorageLayerInterface;
HINT: consider using .StorageLayer().NodeID() instead.

pkg/testutils/physicalplanutils/fake_resolver.go:28: (FakeResolverForTestCluster)
	NOTICE: .AdvRPCAddr() called via implicit interface ApplicationLayerInterface;
HINT: consider using .ApplicationLayer().AdvRPCAddr() instead.
TIP: consider replacing the test server initialization from:
    ts, ... := serverutils.StartServer(t, ...)
    defer ts.Stopper().Stop(...)
to:
    srv, ... := serverutils.StartServer(t, ...)
    defer srv.Stopper().Stop(...)
    ts := srv.ApplicationLayer()

See also: https://go.crdb.dev/p/testserver-and-cluster-virtualization
    logic.go:4186: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/c6b5c27da83706b27606f8db0382a359/logTestLogic_refcursor1288035800
--- FAIL: TestLogic_refcursor (5.42s)
=== RUN   TestLogic_refcursor/functions
    logic.go:2920: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8355/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/refcursor:639: SELECT lag(x) OVER w, lead(x) OVER w, first_value(x) OVER w, last_value(x) OVER w FROM t WINDOW w AS ();
        expected:
            NULL  baz   foo  baz
            foo   NULL  foo  baz
            
        but found (query options: "rowsort" -> ignore the following ordering of rows) :
            NULL  foo   baz  foo
            baz   NULL  baz  foo
            
    logic.go:2177: 
         /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8355/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/refcursor:645: too many errors encountered, skipping the rest of the input
[01:39:39] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8355/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/refcursor with config fakedist-disk: 152 tests, 2 failures
[01:39:39] --- total progress: 152 statements
--- total: 152 tests, 2 failures
    --- FAIL: TestLogic_refcursor/functions (0.01s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_refcursor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-33168

--- a/pkg/sql/logictest/testdata/logic_test/refcursor
+++ b/pkg/sql/logictest/testdata/logic_test/refcursor
@@ -636,10 +636,10 @@ SELECT max(y) FROM t;
 
 # Any-type-matching window functions are allowed.
 query TTTT rowsort
-SELECT lag(x) OVER w, lead(x) OVER w, first_value(x) OVER w, last_value(x) OVER w FROM t WINDOW w AS ();
+SELECT lag(x) OVER w, lead(x) OVER w, first_value(x) OVER w, last_value(x) OVER w FROM t WINDOW w AS (ORDER BY x);
 ----
-NULL  baz   foo  baz
-foo   NULL  foo  baz
+NULL  foo   baz  baz
+baz   NULL  baz  foo
 
 # TODO(#113651): Window functions should work with REFCURSOR[].
 statement error pgcode 42883 pq: unknown signature

